### PR TITLE
Various `std::process` doc improvements.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/libstd/process.rs [hunk 1]
```diff
@@ -91,7 +96,11 @@ impl IntoInner<imp::Process> for Child {
     fn into_inner(self) -> imp::Process { self.handle }
 }
 
-/// A handle to a child process's stdin
+/// A handle to a child process's stdin. This struct is used in the [`stdin`]
```
Comment (GuillaumeGomez) [inline src/libstd/process.rs]: I'm not an english native but I'm pretty sure you don't write "process's" but "process'". But maybe I'm wrong so please correct me if this is the case.

**Comment by GuillaumeGomez** [comment]
For such small changes, I think one commit is more than enough.

File: src/libstd/process.rs (hunk 1 above)
Comment (frewsxcv) [inline src/libstd/process.rs]: I would personally say process's, but I might be wrong and English is weird.

Comment (GuillaumeGomez) [inline src/libstd/process.rs]: Ah. It doesn't help. :laughing:

Comment (apasel422) [inline src/libstd/process.rs]: `process's` is correct.

Comment (GuillaumeGomez) [inline src/libstd/process.rs]: Then my bad!

**Comment by frewsxcv** [comment]
Squashed.

**Comment by frewsxcv** [comment]
Error looks unrelated?

**Comment by GuillaumeGomez** [comment]
Strange...

@bors: retry

**Comment by frewsxcv** [comment]
```
Linkcheck stage2 (i686-pc-windows-gnu)
std\process\struct.Child.html:95: broken link - std\process\or other functions that wrap around it
thread 'main' panicked at 'found some broken links', C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\src\tools\linkchecker\main.rs:54
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

If anyone has any idea which of my links is broken, let me know. I don't have enough time right now to look into it, but it's not obvious to me which is broken.

**Comment by steveklabnik** [comment]
It's not clear to me either :/

File: src/libstd/process.rs [hunk 2]
```diff
@@ -48,13 +49,17 @@ use sys_common::{AsInner, AsInnerMut, FromInner, IntoInner};
...
-/// [`Drop`](../../core/ops/trait.Drop.html) for child processes, so if you
-/// do not ensure the `Child` has exited then it will continue to run, even
-/// after the `Child` handle to the child process has gone out of scope.
+/// Take note that there is no implementation of [`Drop`] for child processes,
+/// so if you do not ensure the `Child` has exited then it will continue to
+/// run, even after the `Child` handle to the child process has gone out of
+/// scope.
 ///
-/// Calling `wait` (or other functions that wrap around it) will make the
+/// Calling [`wait`] (or other functions that wrap around it) will make the
```
Comment (retep998) [inline src/libstd/process.rs]: It is this link which is broken.

Comment (frewsxcv) [inline src/libstd/process.rs]: How is this link broken?

Comment (frewsxcv) [inline src/libstd/process.rs]: Does it think `(or other functions that wrap around it)` is the `href`? Is there a way around this if so?

Comment (alexcrichton) [inline src/libstd/process.rs]: Yeah that's how markdown is interpreting it, you may be able to escape it via `\(` I believe

Comment (frewsxcv) [inline src/libstd/process.rs]: I just force pushed. Changed it to:

```
[`wait`][`wait`]
```

**Pull request merged**